source/AMDSBufferGroup.cpp

```diff
@@ -83,15 +94,23 @@ void AMDSBufferGroup::populateData(AMDSClientRelativeCountPlusCountDataRequest*
 	int count = clientDataRequest->count();
 
 	int startIndex = dataHolders_.count() - 1 - relativeCount;
-	int endIndex = dataHolders_.count() - 1 - relativeCount + count;
+	populateData(clientDataRequest, startIndex, startIndex+count);
+}
 
-	startIndex = qMax(0, startIndex);
-	endIndex = qMin(endIndex, dataHolders_.count() - 1);
+void AMDSBufferGroup::populateData(AMDSClientStartTimeToEndTimeDataRequest* clientDataRequest)
+{
+	QDateTime startTime = clientDataRequest->startTime();
+	QDateTime endTime = clientDataRequest->endTime();
 
-	for (int iCurrent = startIndex; iCurrent < endIndex; iCurrent++)
-	{
-		AMDSDataHolder* dataHolder = dataHolders_[iCurrent];
-		clientDataRequest->appendData(dataHolder);
+	int startIndex = lowerBound(startTime);
```

Comment:
I had a quick look at it, but how expensive is calling `lowerBound()`?  It looks like it's almost like a quick sort.